Fix up grammar and wording in WebContainer's metatype.properties file

volosied opened the pull request:
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".


Small improvements, grammar and clarity, to these property descriptions.

volosied committed:
Fix up grammar and wording

volosied commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -162,7 +162,7 @@ servletDestroyWaitTime.desc=Wait time in seconds for an active request to comple
 servletPathForDefaultMapping.desc=Set the servlet path value to the request URI minus the context path. The path information is null when a servlet is used as a default mapping. The default value is true for version 4.0 or later of the servlet feature. It is false for other servlet features. When mapping is to the /* pattern, the servlet path is empty and the path information starts with a leadin ...
 
-throwExceptionWhenUnableToCompleteOrDispatch.name=Throw exception for not complete or dispatch asyn
+throwExceptionWhenUnableToCompleteOrDispatch.name=Throw exception for not complete or dispatch async
```
Maybe `Throw exception for not completed or dispatched async request`?

volosied commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -172,22 +172,22 @@ stopAppStartUponListenerException.name=Stop application for unhandled listener e
 
 allowQueryParamWithNoEqual.name=Allow query parameters with no value
-allowQueryParamWithNoEqual.desc=When this property is set to true, if the query parameter in a URL contains only the string name, the server returns an empty string as the value for the request.getParameter(name) query. When this property is set to false, the server returns null as the value for the request.getParameter(name) query. The default value is false, but is true for servlet 5.0 and high ...
+allowQueryParamWithNoEqual.desc=When this property is set to true, if the query parameter in a URL contains only the string name, the server returns an empty string as the value for the request.getParameter(name) query. When this property is set to false, the server returns null as the value for the request.getParameter(name) query. The default value is false but is true for servlet 5.0 and highe ...
```
Should these change too?  I don't like when, if statements. 

```=When this property is set to true, if the query parameter```  Maybe if should be moved to the end of the sentence?

pmd1nh commented:
You will also need to tag doc team for their review.

dmuelle commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -18,10 +18,10 @@ web.container.app.name=Web Container
 
 mimeTypes.name=MIME types
-mimeTypes.desc=A list of mappings between a file types and MIME types to use when loading files from this application, for instance txt=text/plain.
+mimeTypes.desc=A list of mappings between a file types and MIME types to use when loading files from this application, for instance, txt=text/plain.
```
if possible, avoid elliptical clauses like "when loading..." as they dont have a clear subject and can be difficult to translate. eg

"A list of mappings between a file types and MIME types to use when you load files..."

I dont know whats doing the loading here so "you" is probably not correct

volosied replied:
Take a look at the new wording :)

dmuelle commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -42,7 +42,7 @@ disallowServeServletsByClassName.name=Disallow serve servlet by class name
 
 doNotServeByClassName.name=Do not serve by class name
-doNotServeByClassName.desc=A semi-colon delimited list of classes to be completely disallowed from being served by classname. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.donotservebyclassname.
+doNotServeByClassName.desc=A semi-colon delimited list of classes to be completely disallowed from being served by class name. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.donotservebyclassname.
```
this is a little unclear-what does "by class name" modify? Also, unless there is a functional difference between "separated" and "delimited", use one term consistently


```suggestion
doNotServeByClassName.desc=A semicolon-separated list of classes by class name that are prohibited from being served. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.donotservebyclassname.
```

dmuelle commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -57,13 +57,13 @@ httpsIndicatorHeader.name=HTTPS indicator header
 
 exposeWebInfOnDispatch.name=Expose web info on dispatch
-exposeWebInfOnDispatch.desc=When this property is set to true, a servlet can access static files in the WEB-INF directory. When this property is set to false, which is the default, a servlet cannot access static files in the WEB-INF directory.
+exposeWebInfOnDispatch.desc=When this property is true, a servlet can access static files in the WEB-INF directory. When this property is set to false, the default, servlets cannot access static files in the WEB-INF directory.
```
```suggestion
exposeWebInfOnDispatch.desc=When this property is set to true, a servlet can access static files in the WEB-INF directory. When this property is set to false, which is the default, servlets cannot access static files in the WEB-INF directory.
```

dmuelle commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -57,13 +57,13 @@ httpsIndicatorHeader.name=HTTPS indicator header
 
 fileWrapperEvents.name=File wrapper events
-fileWrapperEvents.desc=Web container will generate SMF and PMI data when  serving the static files. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.fileWrapperEvents.
+fileWrapperEvents.desc=Web container will generate SMF and PMI data when serving the static files. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.fileWrapperEvents.
```
```suggestion
fileWrapperEvents.desc=The web container generates SMF and PMI data when it serves static files. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.fileWrapperEvents.
```

dmuelle commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -103,7 +103,7 @@ enableDefaultIsElIgnoredInTag.name=Evaluate isELIgnored behavior in JSP tag file
 
 parseUtf8PostData.name=Parse UTF-8 post data
-parseUtf8PostData.desc=Web container will detect non URL encoded UTF-8 post data and include it in the parameter values. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.parseutf8postdata.
+parseUtf8PostData.desc=Web container will detect non-URL encoded UTF-8 post data and include it in the parameter values. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.parseutf8postdata.
```
```suggestion
parseUtf8PostData.desc=The web container detects non-URL encoded UTF-8 post data and include it in the parameter values. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.parseutf8postdata.
```

dmuelle commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -130,10 +130,10 @@ asyncMaxSizeTaskPool.name=Async servlet maximum pool size
 
 asyncTimeoutDefault.name=Default async timeout
-asyncTimeoutDefault.desc=Async servlet timeout value used when a timeout value has not been explcitly specified. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.asynctimeoutdefault.
+asyncTimeoutDefault.desc=Async servlet timeout value used when a timeout value has not been explicitly specified. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.asynctimeoutdefault.
```
```suggestion
asyncTimeoutDefault.desc=Async servlet timeout value that is used when a timeout value is not explicitly specified. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.asynctimeoutdefault.
```

dmuelle commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -145,12 +145,12 @@ emptyServletMappings.name=Return empty servlet mappings
 
 deferServletRequestListenerDestroyOnError.name=Defer servlet request listener destroy
-deferServletRequestListenerDestroyOnError.desc=Toggle if you want to defer ServletRequestListener destroy when there is an error serving the request. The default value is false, but is true for servlet 5.0 and later. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.deferServletRequestListenerDestroyOnError.
+deferServletRequestListenerDestroyOnError.desc=Toggle if you want to defer ServletRequestListener destroy when there is an error serving the request. The default value is false but is true for servlet 5.0 and later. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.deferServletRequestListenerDestroyOnError.
```
```suggestion
deferServletRequestListenerDestroyOnError.desc=Enable this property to defer ServletRequestListener destroy method call when there is an error serving the request. In servlet-5.0 and later, the default value is true. In servlet-4.0 and earlier, the default is false. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.deferServletRequestListenerDestroyOnError.
```

dmuelle commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -145,12 +145,12 @@ emptyServletMappings.name=Return empty servlet mappings
 
 allowExpressionFactoryPerApp.name=Allow expression factory per application
-allowExpressionFactoryPerApp.desc=Toggle to load the ExpressionFactory that is set by the application. Enable this custom property if you are using a custom EL implementation (for example, JUEL) that needs to set its own ExpressionFactory.
+allowExpressionFactoryPerApp.desc=Toggle to load the ExpressionFactory that is set by the application. Enable this custom property if you use a custom EL implementation (for example, JUEL) that needs to set its own ExpressionFactory.
```
```suggestion
allowExpressionFactoryPerApp.desc=Enable this property to load the ExpressionFactory that is set by the application. Enable this custom property if you use a custom EL implementation (for example, JUEL) that needs to set its own ExpressionFactory.
```

dmuelle commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -162,7 +162,7 @@ servletDestroyWaitTime.desc=Wait time in seconds for an active request to comple
 servletPathForDefaultMapping.desc=Set the servlet path value to the request URI minus the context path. The path information is null when a servlet is used as a default mapping. The default value is true for version 4.0 or later of the servlet feature. It is false for other servlet features. When mapping is to the /* pattern, the servlet path is empty and the path information starts with a leadin ...
 
-throwExceptionWhenUnableToCompleteOrDispatch.name=Throw exception for not complete or dispatch asyn
+throwExceptionWhenUnableToCompleteOrDispatch.name=Throw exception for not complete or dispatch async
```
```suggestion
throwExceptionWhenUnableToCompleteOrDispatch.name=Throw an exception for an incomplete or dispatch async request.
```

volosied replied:
Outdated to to my push, but I made a new suggestion further down.

dmuelle commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -172,22 +172,22 @@ stopAppStartUponListenerException.name=Stop application for unhandled listener e
 
 allowQueryParamWithNoEqual.name=Allow query parameters with no value
-allowQueryParamWithNoEqual.desc=When this property is set to true, if the query parameter in a URL contains only the string name, the server returns an empty string as the value for the request.getParameter(name) query. When this property is set to false, the server returns null as the value for the request.getParameter(name) query. The default value is false, but is true for servlet 5.0 and high ...
+allowQueryParamWithNoEqual.desc=When this property is set to true, if the query parameter in a URL contains only the string name, the server returns an empty string as the value for the request.getParameter(name) query. When this property is set to false, the server returns null as the value for the request.getParameter(name) query. The default value is false but is true for servlet 5.0 and highe ...
```
```suggestion
allowQueryParamWithNoEqual.desc=When this property is set to true, if the query parameter in a URL contains only the string name, the server returns an empty string as the value for the request.getParameter(name) query. When this property is set to false, the server returns null as the value for the request.getParameter(name) query. In servlet-5.0 and later, the default value is true. In servlet-4.0 and earlier, the default is false.
```

volosied replied:
Also outdated, but I re-made the suggestion further down.

dmuelle commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -172,22 +172,22 @@ stopAppStartUponListenerException.name=Stop application for unhandled listener e
 
 setHtmlContentTypeOnError.name=Disable text/html content type on error responses
-setHtmlContentTypeOnError.desc=When this property is false, the webcontainer will not set the response's content type header during the error handling process. An application is responsible to set the response's content type. The default value (true) sets the content type to "text/html".
+setHtmlContentTypeOnError.desc=When this property is false, the webcontainer will not set the response's content-type header during the error handling process. An application is responsible for setting the response's content type. The default value (true) sets the content type to "text/html".
```
```suggestion
setHtmlContentTypeOnError.desc=When this property is false, the web container does not set the response content-type header during the error handling process. An application is responsible for setting the response content type. The default value (true) sets the content type to "text/html".
```

dmuelle commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -199,4 +199,4 @@ set400SCOnTooManyParentDirs.name=Set 400 for invalid path traversal request
 
 allowAbsoluteFileNameForPartWrite.name=Write part files to absolute file locations
-allowAbsoluteFileNameForPartWrite.desc=When this property is set to true, the runtime treats the filename argument on the part.write method call as an absolute location, rather than saving the filename arguement under the temporary location. In servlet-6.0 and earlier, the default value is false. In servlet-6.1 and later, the default value is true.
+allowAbsoluteFileNameForPartWrite.desc=When this property is set to true, the runtime treats the filename argument on the part.write method call as an absolute location, rather than saving the filename argument under the temporary location. In servlet-6.0 and earlier, the default value is false. In servlet-6.1 and later, the default value is true.
```
```suggestion
allowAbsoluteFileNameForPartWrite.desc=When this property is set to true, the runtime treats the filename argument on the part.write method call as an absolute location, rather than saving the filename argument under the temporary location. In servlet-6.1 and later, the default value is true. In servlet-6.0 and earlier, the default value is false. 
```

dmuelle reviewed (changes requested):
ID review

volosied committed:
Address review comments

Apply suggestions from code review

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

Apply suggestions from code review

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

Update dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

Update dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties

Update dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties

Co-authored-by: David Mueller <48686014+dmuelle@users.noreply.github.com>

dmuelle commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -103,7 +103,7 @@ enableDefaultIsElIgnoredInTag.name=Evaluate isELIgnored behavior in JSP tag file
 
 parseUtf8PostData.name=Parse UTF-8 post data
-parseUtf8PostData.desc=Web container will detect non URL encoded UTF-8 post data and include it in the parameter values. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.parseutf8postdata.
+parseUtf8PostData.desc=The web container detects non-URL encoded UTF-8 post data and include it in the parameter values. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.parseutf8postdata.
```
```suggestion
parseUtf8PostData.desc=The web container detects non-URL encoded UTF-8 post data and includes it in the parameter values. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.parseutf8postdata.
```

dmuelle commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -145,12 +145,12 @@ emptyServletMappings.name=Return empty servlet mappings
 
 deferServletRequestListenerDestroyOnError.name=Defer servlet request listener destroy
-deferServletRequestListenerDestroyOnError.desc=Toggle if you want to defer ServletRequestListener destroy when there is an error serving the request. The default value is false, but is true for servlet 5.0 and later. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.deferServletRequestListenerDestroyOnError.
+deferServletRequestListenerDestroyOnError.desc=Enable this property to defer ServletRequestListener destroy method call when there is an error serving the request. In servlet-5.0 and later, the default value is true. In servlet-4.0 and earlier, the default is false. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.deferServletRequestListenerDestroyO ...
```
```suggestion
deferServletRequestListenerDestroyOnError.desc=Enable this property to defer the ServletRequestListener destroy method call when there is an error serving the request. In servlet-5.0 and later, the default value is true. In servlet-4.0 and earlier, the default is false. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.deferServletRequestListenerDestroyOnError.
```

dmuelle commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -172,31 +172,31 @@ stopAppStartUponListenerException.name=Stop application for unhandled listener e
 
 allowQueryParamWithNoEqual.name=Allow query parameters with no value
-allowQueryParamWithNoEqual.desc=When this property is set to true, if the query parameter in a URL contains only the string name, the server returns an empty string as the value for the request.getParameter(name) query. When this property is set to false, the server returns null as the value for the request.getParameter(name) query. The default value is false, but is true for servlet 5.0 and high ...
+allowQueryParamWithNoEqual.desc=When this property is set to true, the server returns an empty string as the value for the request.getParameter(name) query if the query parameter in the URL only contains the string name. When this property is set to false, the server returns null as the value for the request.getParameter(name) query. The default value is false but is true for servlet 5.0 and high ...
```
```suggestion
allowQueryParamWithNoEqual.desc=When this property is set to true, the server returns an empty string as the value for the request.getParameter(name) query if the query parameter in the URL contains only the string name. When this property is set to false, the server returns null as the value for the request.getParameter(name) query. In servlet-5.0 and later, the default value is true. In servlet-4.0 and earlier, the default is false.
```

dmuelle reviewed (changes requested):
A few more minor suggestions

volosied commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -162,7 +162,7 @@ servletDestroyWaitTime.desc=Wait time in seconds for an active request to comple
 servletPathForDefaultMapping.desc=Set the servlet path value to the request URI minus the context path. The path information is null when a servlet is used as a default mapping. The default value is true for version 4.0 or later of the servlet feature. It is false for other servlet features. When mapping is to the /* pattern, the servlet path is empty and the path information starts with a leadin ...
 
-throwExceptionWhenUnableToCompleteOrDispatch.name=Throw exception for not complete or dispatch asyn
+throwExceptionWhenUnableToCompleteOrDispatch.name=Throw exception for not completed or dispatched async request
```
```suggestion
throwExceptionWhenUnableToCompleteOrDispatch.name=Throw an exception for an incomplete or dispatched async request
```

volosied replied:
@pmd1nh  Let me know if the wording is good here

pmd1nh replied:
LGTM

volosied commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -172,31 +172,31 @@ stopAppStartUponListenerException.name=Stop application for unhandled listener e
 
 allowQueryParamWithNoEqual.name=Allow query parameters with no value
-allowQueryParamWithNoEqual.desc=When this property is set to true, if the query parameter in a URL contains only the string name, the server returns an empty string as the value for the request.getParameter(name) query. When this property is set to false, the server returns null as the value for the request.getParameter(name) query. The default value is false, but is true for servlet 5.0 and high ...
+allowQueryParamWithNoEqual.desc=When this property is set to true, the server returns an empty string as the value for the request.getParameter(name) query if the query parameter in the URL contains only the string name. When this property is set to false, the server returns null as the value for the request.getParameter(name) query. In servlet-5.0 and later, the default value is true. In servlet ...
```
```suggestion
allowQueryParamWithNoEqual.desc=When this property is set to true, if the query parameter in a URL contains only the string name, the server returns an empty string as the value for the request.getParameter(name) query. When this property is set to false, the server returns null as the value for the request.getParameter(name) query. In servlet-5.0 and later, the default value is true. In servlet-4.0 and earlier, the default is false.
```

volosied replied:
@dmuelle  which do you prefer?  I'm leaning on current wording (if is at the end rather than in the middle of the sentence)

dmuelle replied:
@volosied - does `if the query parameter in a URL contains only the string name` also apply if the prop is set to false? If so, it might be more clear to first set that condition in the description... something like

```suggestion
allowQueryParamWithNoEqual.desc=This property determines how the server responds when the query parameter in a URL contains only the string name. When it is set to true, the server returns an empty string as the value for the request.getParameter(name) query.  When it is set to false, the server returns null as the value for the request.getParameter(name) query. In servlet-5.0 and later, the default value is true. In servlet-4.0 and earlier, the default is false.
```

dmuelle commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -172,31 +172,31 @@ stopAppStartUponListenerException.name=Stop application for unhandled listener e
 
 allowQueryParamWithNoEqual.name=Allow query parameters with no value
-allowQueryParamWithNoEqual.desc=When this property is set to true, if the query parameter in a URL contains only the string name, the server returns an empty string as the value for the request.getParameter(name) query. When this property is set to false, the server returns null as the value for the request.getParameter(name) query. The default value is false, but is true for servlet 5.0 and high ...
+allowQueryParamWithNoEqual.desc=This property determines how the server responds when the query parameter in a URL contains only the string name. When it is set to true, the server returns an empty string as the value for the request.getParameter(name) query.  When it is set to false, the server returns null as the value for the request.getParameter(name) query. In servlet-5.0 and later, the defa ...
```
```suggestion
allowQueryParamWithNoEqual.desc=This property determines how the server responds when the query parameter in a URL contains only the string name. When it is set to true, the server returns an empty string as the value for the request.getParameter(name) query. When it is set to false, the server returns null as the value for the request.getParameter(name) query. In servlet-5.0 and later, the default value is true. In servlet-4.0 and earlier, the default is false.
```

dmuelle reviewed (approved):
LGTM thanks

volosied commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=09351816-8692-426b-af20-5e9636b22699) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

volosied commented:
#run-check
